packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -339,25 +339,30 @@ class AppDomain extends Domain {
       );
     }
 
+    bool supportsRestart = hotMode ? device.supportsHotMode : device.supportsRestart;
```

Comment:
An isolate can hit an unhandled exception error at anytime, when that happens reload needs to be disabled until a full restart occurs.